test/util/application.go

```diff
@@ -11,6 +11,11 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+func PrepareInClusterApplication(t *testing.T, objects ...runtime.Object) application.Application {
```

Comment:
minor - It might seem a bit unclear at first to have functions named `PrepareInClusterApplication` and `PrepareInClusterApp` , but I see that you wanted to have one which returns only the `Application` type without the `fakeClient` , not sure how we could improve this.

Reply:
Yes, I know that the names can be a bit confusing. I believe that I'll be able to improve them or get rid of one of them in the following PRs.